core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala

```diff
@@ -311,25 +311,53 @@ abstract class ApiDocDirective(name: String)
 }
 
 object ApiDocDirective {
-  /** This relies on the naming convention of packages being all-ascii-lowercase (which is rarely broken), numbers and underscore. */
-  def packageDotsToSlash(s: String) = s.replaceAll("(\\b[a-z][a-z0-9_]*)\\.", "$1/")
+  /**
+   * Converts package dot notation to a path, separated by '/'
+   * Allow all valid java characters and java numbers to be used, according to the java lang spec. In order to
+   * be allow to convert `package.Outer.Inner` in scala to the `packge/Outer$$Inner`, `strictPackageIdent` has
+   * be set to `true`.
+   * @param s package or full qualified class name to be converted.
+   * @param packageNameStyle  `startWithLowercase` requires subpackages to start with a lowercase character
+   *                          (which is the default to be backwards compatible).
+   *                          `startWithAnycase` allows packages to start with uppercase also.
+   * @return Resulting path.
+   */
+  def packageDotsToSlash(s: String, packageNameStyle: String = "startWithLowercase") =
```

Comment:
This default isn't used anywhere, right?
```suggestion
  def packageDotsToSlash(s: String, packageNameStyle: String) =
```

Reply:
Yes it is not explicitly checked. In the first thought I wanted to define it as simple flag, but using it as a style is more flexile. Was also wondering if it makes sense to add the old style where ascii characters were allowed only, but skipped this. Do you think it is worse to add?

Reply:
I don't think we need the old behavior anymore :+1: